### PR TITLE
N3DS Touchscreen toggle 

### PR DESCRIPTION
**Pull request by Narwher**
The numpad on the 3DS touchscreen may not be used for some games. This change makes it so that it is hidden by default upon startup (the LCD is turned off as well) and is activated by touch input.

There is currently no way to turn it back off. Almost all button bindings are used by the interpreter, so it would likely have to be a something involving the touchscreen again. The '0' button on the numpad occupies the width of two normal keys - perhaps dividing it in half and adding a button for LCD toggle would work?

Side note: I also experimented with changing the sample rate for 3DS audio but have since changed it back. As I have not cherry-picked it all of these commits are in the PR. Please only examine commits from 18 June onwards.

Thanks for your consideration!

### CODE REVIEW

**Comment by Ghabry** [comment]
About a way to configure this / turn it off again:
By default of sounds fine.

We currently also work on a settings UI. It could also contain an option for toggling the LCD later.

**Comment by Narwher** [comment]
With [8c6a6c3](https://github.com/EasyRPG/Player/pull/2252/commits/8c6a6c3ee2fb39e720bbf09db811472db31f432e) I made a rough draft for a possible extra button on the numpad overlay, to turn off the display. There's no function tied to it yet though.

keyboard

File: src/platform/3ds/3ds_ui.cpp [hunk 1]
```diff
@@ -220,6 +227,13 @@ void CtrUi::ProcessEvents() {
 		touch_y = pos.py;
 
 		keys[keys_tbl[col + (row * 4)]] = true;
+
+		if (!touchscreen) {
+			touchscreen = true;
+			gspLcdInit();
```
Comment (carstene1ns) [inline src/platform/3ds/3ds_ui.cpp:233]: Please with error checking:
```suggestion
			if (R_SUCCEEDED(gspLcdInit())) {
				GSPLCD_PowerOnBacklight(GSPLCD_SCREEN_BOTTOM);
				gspLcdExit();
			}
```

File: src/platform/3ds/3ds_ui.cpp [hunk 2]
```diff
@@ -78,13 +80,19 @@ CtrUi::CtrUi(int width, int height)
 		osSetSpeedupEnable(true);
 	}
 
+	// Turn off bottom screen initially
```
Comment (carstene1ns) [inline src/platform/3ds/3ds_ui.cpp:83]: why?

File: src/platform/3ds/3ds_ui.cpp [hunk 3]
```diff
@@ -277,49 +291,46 @@ void CtrUi::UpdateDisplay() {
...
-		u8 pos_x = col * button_width;
-		u8 pos_y = row * button_height;
-
-		// "0" is handled specially
-		u8 draw_width = button_width;
-		if (col < 2 && row == 3) {
-			draw_width *= 2;
-			if (col == 1)
-				pos_x = 0;
+		C2D_SceneBegin(bottom_screen);
```
Comment (carstene1ns) [inline src/platform/3ds/3ds_ui.cpp:294]: The indentation is completely off here.

**Comment by fdelapena** [comment]
Jenkins CI: ok to test

**Comment by troybonneau** [comment]
This is marked for 7.0, so I was wondering whether or not this needed a bit more work before it gets in. I think I could probably finish up the graphics/code if no one is available.

**Comment by Ghabry** [comment]
#2617 replaced this one

**Pull request closed without merging**